### PR TITLE
Close PrestoSparkService on the Driver

### PR DESCRIPTION
**Pull request by arhimondr**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java [hunk 1]
```diff
@@ -480,7 +479,7 @@ private TestingPrestoSparkTaskExecutorFactoryProvider(String instanceId)
         }
 
         @Override
-        public IPrestoSparkTaskExecutorFactory get(SparkProcessType processType)
+        public IPrestoSparkTaskExecutorFactory get()
```
Comment (wenleix) [inline presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java:482]: why no longer need `processType` ?

File: presto-spark-launcher/src/main/java/com/facebook/presto/spark/launcher/PrestoSparkRunner.java [hunk 2]
```diff
@@ -122,48 +134,64 @@ private static IPrestoSparkServiceFactory createServiceFactory(File directory)
...
-        private static String currentPackagePath;
-        private static Map<String, String> currentConfigProperties;
-        private static Map<String, Map<String, String>> currentCatalogProperties;
-        private static SparkProcessType currentProcessEnvironment;
+    private static String getPackagePath(PackageSupplier packageSupplier)
+    {
+        return checkDirectory(packageSupplier.getPrestoSparkPackageDirectory()).getAbsolutePath();
+    }
 
+    private static class DistributionBasedPrestoSparkTaskExecutorFactoryProvider
```
Comment (wenleix) [inline presto-spark-launcher/src/main/java/com/facebook/presto/spark/launcher/PrestoSparkRunner.java:155]: This class name is very wordy.. 😃

File: presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java (hunk 1 above)
Comment (arhimondr) [reply presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java:482]: PrestoSparkTaskExecutorFactoryProvider is only used to create tasks on executor, so it is always `EXECUTOR`.

File: presto-spark-launcher/src/main/java/com/facebook/presto/spark/launcher/PrestoSparkRunner.java (hunk 2 above)
Comment (arhimondr) [reply presto-spark-launcher/src/main/java/com/facebook/presto/spark/launcher/PrestoSparkRunner.java:155]: Yeah, unfortunately i didn't find a better name. We can think of a better name later.

**Pull request merged**